### PR TITLE
feat: Add insert mode mappings as well

### LINKED ISSUE #178: Feature request: submit chat in insert mode
The submit mapping (default `<CR>`) only works in normal mode.
Would be great if there was an option to have it work in insert mode too.
I've tried creating a mapping such as `<C-g>` but it only works in normal mode.
Having to exit insert mode when in the middle of a back and forth feels a bit cumbersome at times.

Thanks so much for building such an awesome plugin!!

### PR DESCRIPTION
**Pull request by deathbeam**
Closes #178

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -263,13 +263,13 @@ Also see [here](/lua/CopilotChat/config.lua):
...
 
-  -- default mappings
+  -- default mappings (in tables first is normal mode, second is insert mode)
   mappings = {
-    close = 'q',
-    reset = '<C-l>',
     complete = '<Tab>',
-    submit_prompt = '<CR>',
-    accept_diff = '<C-y>',
+    close = { 'q', '<C-c>' },
```
[the remark this answers is not in the document]
Comment (deathbeam) [reply README.md:269]: hmm yea, maybe I could just do all of them like that so its super verbose i guess

[the remark this answers is not in the document]
Comment (deathbeam) [reply README.md]: Alright should be updated. But this also means we no longer support the simple string format. Which imo is fine

File: README.md [hunk 2]
```diff
@@ -263,16 +263,37 @@ Also see [here](/lua/CopilotChat/config.lua):
     zindex = 1, -- determines if window is on top or below other floating windows
   },
 
-  -- default mappings
+  -- default mappings (in tables first is normal mode, second is insert mode)
```
Comment (jellydn) [inline README.md:266]:
```suggestion
  -- default mappings
```

**Review by jellydn — APPROVED** [review]
Thanks @deathbeam LGTM!

**Pull request merged**